### PR TITLE
Tweak `no-unused-vars` rule

### PR DESCRIPTION
**Pull request by LostCrew**
- Raise errors when arguments are unused
- Allow to use rest operator to discard unwanted siblings

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -52,7 +52,11 @@ module.exports = {
     'no-unexpected-multiline': 'error',
     'no-unreachable': 'error',
     'no-unsafe-finally': 'error',
-    'no-unused-vars': ['error', {vars: 'all', args: 'none'}],
+    'no-unused-vars': ['error', {
+      vars: 'all',
+      args: 'after-used',
+      ignoreRestSiblings: true
+    }],
```
Comment (tschaub) [inline index.js]: It looks like Prettier wants this to be like:
```js
    'no-unused-vars': [
      'error',
      {
        vars: 'all',
        args: 'after-used',
        ignoreRestSiblings: true
      }
    ],
```

Looks like you could also just do `'no-unused-vars': ['error', {ignoreRestSiblings: true}]`.

**Comment by tschaub** [comment]
This change sounds good to me.  It looks like the only change here is to ignore rest siblings (default is [`ignoreRestSiblings: false`](https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings)) -
 so not necessarily a breaking change.  After merging we could bump the minor version.

File: index.js (hunk 1 above)
Comment (LostCrew) [reply index.js]: Sorry, I haven't install dependencies for development on this project. 
Your suggested value, is in fact, how it is set in one of our private repository. I will update it accordingly.

**Comment by LostCrew** [comment]
Sounds great!

**Pull request merged**